corehq/ex-submodules/casexml/apps/phone/restore.py

```diff
@@ -93,17 +97,72 @@ def extend(self, iterable):
         for element in iterable:
             self.append(element)
 
-    def __str__(self):
+    def compose(self):
+        raise NotImplemented()
+
+
+class FileRestoreResponse(RestoreResponse):
+
+    BODY_TAG_SUFFIX = '-body'
+
+    def __init__(self, username=None, items=False):
+        super(FileRestoreResponse, self).__init__(username, items)
+        self.filename = path.join(settings.RESTORE_PAYLOAD_DIR, uuid4().hex)
+        self.extention = 'xml'
+
+        self.response_body = FileIO(self.get_filename(self.BODY_TAG_SUFFIX), 'w+')
+        self.response = FileIO(self.get_filename(), 'w+')
+
+    def get_filename(self, suffix=None):
+        return "{filename}{suffix}.{ext}".format(
+            filename=self.filename,
+            suffix=suffix or '',
+            ext=self.extention
+        )
+
+    def __add__(self, other):
+        if not isinstance(other, FileRestoreResponse):
+            raise NotImplemented()
+
+        response = FileRestoreResponse(self.username, self.items)
+        response.num_items = self.num_items + other.num_items
+
+        self.response_body.seek(0)
+        other.response_body.seek(0)
+
+        shutil.copyfileobj(self.response_body, response.response_body)
+        shutil.copyfileobj(other.response_body, response.response_body)
```

Comment:
I was wondering about that as well. I tried to touch the algorithm as little as possible, but it seems like we could add a parameter to the `payload_fn` that optionally takes in an existing `RestoreResponse`. That way we don't have to add the two together here: https://github.com/benrudolph/commcare-hq/blob/ota-file-161809/corehq/ex-submodules/casexml/apps/phone/restore.py#L451